### PR TITLE
Prepare 1.0.0 release

### PR DESCRIPTION
**Pull request by pirgeo**
* Bump package version to 1.0.0 & update readme
* Update GH actions to latest versions
* Add `all-passed` check

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -9,30 +9,30 @@ on:
...
     name: Java ${{ matrix.java-version }} on ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up JDK ${{ matrix.java-version }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
+          distribution: 'temurin'
```
Comment (arminru) [inline .github/workflows/gradle.yml:33]: Please add a comment on why we need this particular one.

File: README.md [hunk 2]
```diff
@@ -1,41 +1,25 @@
...
-// settings.gradle:
-sourceControl {
-    gitRepository("https://github.com/dynatrace-oss/opentelemetry-metric-java.git") {
-        producesModule("com.dynatrace.opentelemetry.metric:dynatrace")
-    }
-}
-
-// build.gradle:
-// use the name of a specific tag from https://github.com/dynatrace-oss/opentelemetry-metric-java/tags
-def dynatraceMetricsExporterVersion = "v0.5.0"
```
Comment (arminru) [inline README.md:30]: This would need to be kept for now, right?

Comment (pirgeo) [reply README.md]: Changed it back :)

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (pirgeo) [reply .github/workflows/gradle.yml:35]: I added a comment. With the latest setup-java action, you need to specify a distribution, and Temurin seems to be suggested by a few sites I checked, but there is no "definite right answer"...

**Review by joaopgrassi — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -51,7 +46,7 @@ More information on setting up API access using tokens can be found [in the docu
 DynatraceMetricExporter exporter =
     DynatraceMetricExporter.builder()
         .setUrl("https://{your-environment-id}.live.dynatrace.com/api/v2/metrics/ingest")
-        .setApiToken({YOUR_API_TOKEN}) // read from environment or config
```
Comment (pichlermarc) [inline README.md:54]: I'm wondering if the curly braces could make it clearer that the API token is missing. :thinking:

File: README.md [hunk 4]
```diff
@@ -27,15 +24,13 @@ sourceControl {
...
 // use the name of a specific tag from https://github.com/dynatrace-oss/opentelemetry-metric-java/tags
-def dynatraceMetricsExporterVersion = "v0.5.0"
+def dynatraceMetricsExporterVersion = "v1.0.0"
 
 dependencies {
     implementation("com.dynatrace.opentelemetry.metric:dynatrace:${dynatraceMetricsExporterVersion}")
 }
 ```
 
-Gradle pulls the library in the specified version directly from GitHub and includes it.
```
Comment (pichlermarc) [inline README.md:37]: Just wondering, will the released version be published to maven central? :thinking:

File: README.md [hunk 5]
```diff
@@ -1,15 +1,12 @@
 # Dynatrace OpenTelemetry Metrics Exporter for Java
 
-This exporter allows exporting metrics created using the [OpenTelemetry SDK for Java](https://github.com/open-telemetry/opentelemetry-java)
-directly to [Dynatrace](https://www.dynatrace.com).
+This exporter allows exporting metrics created using the [OpenTelemetry SDK for Java](https://github.com/open-telemetry/opentelemetry-java) directly to [Dynatrace](https://www.dynatrace.com).
+
+It was built against OpenTelemetry SDK version [1.14.0](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.14.0) and should work with any 1.14.+ version.
```
Comment (pichlermarc) [inline README.md:5]:
```suggestion
It was built against OpenTelemetry SDK version [1.14.0](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.14.0) and should work with any 1.14+ version.
```
(can be ignored, just am not sure if we should keep the `.`) :slightly_smiling_face:

**Review by pichlermarc — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (pirgeo) [reply README.md:37]: No, we discussed that internally and will continue with the source dependency. Seems to work fine!

File: README.md (hunk 5 above)
Comment (pichlermarc) [reply README.md:5]: Sounds good, it's just a nit anyway. :slightly_smiling_face: 
I thought that I usually see that written as `1.14+`, but could have been my brain fooling me. :sweat_smile:

Comment (pirgeo) [reply README.md:5]: oh 😮, you are right. I thought I had it right. Really nice catch, updating now.

**Pull request merged**